### PR TITLE
fix(songpal): make patches more resilient

### PR DESCRIPTION
**Pull request by 1fexd**
As already discussed in #2369, I have attempted to make the songpal patches more resilient.

### CODE REVIEW

**Comment by oSumAtrIX** [comment]
Made some changes, please check if everything works as expected

**Comment by 1fexd** [comment]
Looks good. I have also pushed an update to the method name.

File: src/main/kotlin/app/revanced/patches/songpal/badge/fingerprints/ShowNotificationFingerprint.kt [hunk 1]
```diff
@@ -30,7 +30,7 @@ object ShowNotificationFingerprint : MethodFingerprint(
 ) {
     val expectedReference = ImmutableMethodReference(
         "Lcom/google/android/material/bottomnavigation/BottomNavigationView;",
-        "showNotification", // Placeholder.
+        "f", // While the method is called "f" in 9.5.0, this may change in a future version, so it is not used for matching.
```
Comment (oSumAtrIX) [inline src/main/kotlin/app/revanced/patches/songpal/badge/fingerprints/ShowNotificationFingerprint.kt:33]: I don't think "f" is useful, just give the method reference an explaining name such as `showNotification` and comment the version of the app this fingerprint was made for somewhere

**Review by oSumAtrIX — CHANGES_REQUESTED** [review]

**Comment by 1fexd** [comment]
Is this better?

File: src/main/kotlin/app/revanced/patches/songpal/badge/fingerprints/ShowNotificationFingerprint.kt [hunk 2]
```diff
@@ -1,31 +1,37 @@
...
+                    if (it.returnType != returnType) return@any false
+                }
+            }
+            true
+        } ?: false
+    }
+) {
+    val expectedReference = ImmutableMethodReference(
+        "Lcom/google/android/material/bottomnavigation/BottomNavigationView;",
+        "getOrCreateBadge", // This is the non-obfuscated method name, but this is not used for matching since it may change due to R8/Proguard
```
Comment (oSumAtrIX) [inline src/main/kotlin/app/revanced/patches/songpal/badge/fingerprints/ShowNotificationFingerprint.kt:33]:
```suggestion
        "getOrCreateBadge", // Non-obfuscated placeholder method name
```

**Pull request merged**